src/main/scala/rules/Brancher.scala

```diff
@@ -151,17 +154,25 @@ object brancher extends BranchingRules {
         CompletableFuture.completedFuture(Seq(Unreachable()))
       }
 
-    val res = (if (executeThenBranch) {
-      v.symbExLog.markReachable(uidBranchPoint)
-      executionFlowController.locally(s, v)((s1, v1) => {
-        v1.decider.prover.comment(s"[then-branch: $cnt | $condition]")
-        v1.decider.setCurrentBranchCondition(condition, conditionExp)
+    val res = {
+      val thenRes = if (executeThenBranch) {
+          v.symbExLog.markReachable(uidBranchPoint)
+          executionFlowController.locally(s, v)((s1, v1) => {
+            v1.decider.prover.comment(s"[then-branch: $cnt | $condition]")
+            v1.decider.setCurrentBranchCondition(condition, conditionExp)
 
-        fThen(v1.stateConsolidator.consolidateIfRetrying(s1, v1), v1)
-      })
-    } else {
-      Unreachable()
-    }).combine({
+            fThen(v1.stateConsolidator.consolidateIfRetrying(s1, v1), v1)
+          })
+        } else {
+          Unreachable()
+        }
+      if (thenRes.isFatal && !thenRes.isReported && parallelizeElseBranch && s.retryLevel == 0) {
```

Comment:
Why does the condition have the conjuncts`parallelizeElseBranch`? why not `s.parallelizeBranches` instead? Also, I am reading `s.retryLevel == 0` as meaning "there are no more retries left". Is this correct? Given that this comparison appears often in multiple parts of the code, it could make sense to add an aptly named method to `State` like `isLastRetry(): bool`

Reply:
Yes, good idea, I'll add isLastRetry.

Regarding the first point: I think that makes no difference at the end of the day. parallelizeElseBranch may be false even when s.parallelizeBranches is true if one of the branches is not feasible. So before adding the isReported field, branching on parallelizeElseBranch would have meant omitting some unnecessary messages; now, there should be no difference. I'll change it since s.parallelizeBranches might be clearer.